e2e/src/api/specs/person.e2e-spec.ts

```diff
@@ -0,0 +1,176 @@
+import { LoginResponseDto, PersonResponseDto } from '@immich/sdk';
+import { uuidDto } from 'src/fixtures';
+import { errorDto } from 'src/responses';
+import { apiUtils, app, dbUtils } from 'src/utils';
+import request from 'supertest';
+import { beforeAll, beforeEach, describe, expect, it } from 'vitest';
+
+describe('/activity', () => {
+  let admin: LoginResponseDto;
+  let visiblePerson: PersonResponseDto;
+  let hiddenPerson: PersonResponseDto;
+
+  beforeAll(async () => {
+    apiUtils.setup();
+    await dbUtils.reset();
+    admin = await apiUtils.adminSetup();
+  });
+
+  beforeEach(async () => {
+    await dbUtils.reset(['person']);
+
+    [visiblePerson, hiddenPerson] = await Promise.all([
+      apiUtils.createPerson(admin.accessToken, {
+        name: 'visible_person',
+      }),
+      apiUtils.createPerson(admin.accessToken, {
+        name: 'hidden_person',
+        isHidden: true,
+      }),
+    ]);
+
+    const asset = await apiUtils.createAsset(admin.accessToken);
+
+    await Promise.all([
+      dbUtils.createFace({ assetId: asset.id, personId: visiblePerson.id }),
+      dbUtils.createFace({ assetId: asset.id, personId: hiddenPerson.id }),
+    ]);
+  });
+
+  describe('GET /person', () => {
+    beforeEach(async () => {});
+
+    it('should require authentication', async () => {
+      const { status, body } = await request(app).get('/person');
+
+      expect(status).toBe(401);
+      expect(body).toEqual(errorDto.unauthorized);
+    });
+
+    it('should return all people (including hidden)', async () => {
+      const { status, body } = await request(app)
+        .get('/person')
+        .set('Authorization', `Bearer ${admin.accessToken}`)
+        .query({ withHidden: true });
+
+      expect(status).toBe(200);
+      expect(body).toEqual({
+        total: 2,
+        people: [
+          expect.objectContaining({ name: 'visible_person' }),
+          expect.objectContaining({ name: 'hidden_person' }),
+        ],
+      });
+    });
+
+    it('should return only visible people', async () => {
+      const { status, body } = await request(app)
+        .get('/person')
+        .set('Authorization', `Bearer ${admin.accessToken}`);
+
+      expect(status).toBe(200);
+      expect(body).toEqual({
+        total: 2,
```

Comment:
Why is that total 2?

Reply:
The response has total: number and items: Person[] or something like that. This returns both people.

Reply:
Shouldn't it only return the visible person?

Reply:
Idk. Maybe the test case is mislabelled now. I just copied it is as.